### PR TITLE
Add function to add tree

### PR DESCRIPTION
**Pull request by 0xMuluh**
ping #498

### CODE REVIEW

File: R/taxonomy.R [hunk 1]
```diff
@@ -69,6 +69,10 @@
 #' @param use_grepl \code{TRUE} or \code{FALSE}: should pattern matching via
 #'   \code{grepl} be used? Otherwise literal matching is used.
 #'   (default: \code{FALSE})
+#'   
+#'@param MARGIN A character value for selecting if data is merged
+#'   row-wise / for features ('rows') or column-wise / for samples ('cols').
+#'   Must be \code{'rows'} or \code{'cols'}.
```
Comment (TuomasBorman) [inline R/taxonomy.R:71]: This could be hidden. There is not many use-case currently for MARGIN = 2, but good to have support

File: R/taxonomy.R [hunk 2]
```diff
@@ -532,7 +536,7 @@ setGeneric("addHierarchyTree",
 #' @rdname hierarchy-tree
 #' @export
 setMethod("addHierarchyTree", signature = c(x = "SummarizedExperiment"),
-    function(x, ...){
+    function(x, replace = FALSE, ...){
```
Comment (TuomasBorman) [inline R/taxonomy.R]: Also these all could be moved to utils.R

File: R/utils.R [hunk 3]
```diff
@@ -351,6 +351,46 @@
     altExps(x) <- values
     return(x)
 }
+
+# This function can be used to add tree to existing tree
+.add_tree <- function(x, tree, MARGIN, name = "phylo", ...) {
+  
+  MARGIN <- .check_MARGIN(MARGIN)
+  
+  if (MARGIN == 1L) {
```
Comment (TuomasBorman) [inline R/utils.R]: Tidier way is to get function like this https://github.com/microbiome/mia/blob/b627edce620807af20d3ed85c6667f4b8ef8f2ea/R/agglomerate.R#L452

File: R/utils.R [hunk 4]
```diff
@@ -351,6 +351,46 @@
...
+    # Add row tree
+    if (is.null(x@rowTree)) {
```
Comment (TuomasBorman) [inline R/utils.R]: You should not use @. Instead, use rowTreeNames

File: R/utils.R [hunk 5]
```diff
@@ -351,6 +351,46 @@
...
+      x@rowTree <- list(phylo = tree)
+    } else {
+      # Check if a tree with the given name already exists
+      if (name %in% names(x@rowTree)) {
+        warning(paste("A tree named", name, "already exists in the row tree. The new tree will be added with a different name."))
+        x@rowTree <- c(x@rowTree, list(tree))
+      } else {
+        x@rowTree[[name]] <- tree
```
Comment (TuomasBorman) [inline R/utils.R]: I think this does not work

File: R/utils.R [hunk 6]
```diff
@@ -351,6 +351,46 @@
     altExps(x) <- values
     return(x)
 }
+
+# This function can be used to add tree to existing tree
+.add_tree <- function(x, tree, MARGIN, name = "phylo", ...) {
```
Comment (TuomasBorman) [inline R/utils.R]: So:

1. Add all the functionality to .add_tree() function.
2. Check MARGIN and name
3. Get correct functions based on MARGIN
4. Check if tree name. If it does exist, give warning
5. Add tree to object

changeTree() seems not to work when the tree does not already exist with that name.

**Comment by TuomasBorman** [comment]
Can you check: it seems that row can have only one link in rowLinks. This means that no extra tree can be added (then there would be more than one linkage between single row and trees). There can be multiple trees if the dataset is divided into several trees. 

--> Can you check if that is true

**Comment by 0xMuluh** [comment]
Load sample data
```
> library(mia)
> data(GlobalPatterns, package="mia")
> tse <- GlobalPatterns
> tse <- transformAssay(tse, MARGIN = "samples", method="relabundance")
```
Check the rowLinks slot
`>row_links <- rowLinks(tse)`

Count the number of links per row
`> num_links_per_row <- sapply(row_links, length)`

Check if any row has multiple links
```
> has_multiple_links <- any(num_links_per_row > 1)
> has_multiple_links
TRUE
```

row seems to have multiple links.

**Comment by TuomasBorman** [comment]
Your example does not show that. If you check rowLinks(tse):

1. Row in rowLinks() represent a row in TreeSE. For instance, row number 1 in TreeSE is a row number 1 in rowLinks
2. nodeNum tells which node represents the row in rowTree. In rowTree, multiple rows of TreeSE can be represented by a single node.

**Comment by TuomasBorman** [comment]
But the question was:

1. If we add new rowTree, it must be linked with rows. 
2. Can there be more rows in rowLinks than there are rows in TreeSE. For instance in GlobalPatterns dataset there would be then almost 40,000 rows in rowLinks versus ~ 20,000 rows in TreeSE

--> I think this is not possible to do. I beliave the linkage between rows in TreeSE and rows in rowLinks is "hard" that way that each row in TreeSE is linked to corresponding index in rowLinks (but I might be wrong)

If that is the case, then we can just close this PR and issue --> the thing has be been checked

**Comment by TuomasBorman** [comment]
Any updates on this?

**Comment by TuomasBorman** [comment]
updates?

**Comment by 0xMuluh** [comment]
Not yet. Still unable to clarify if TreeSE can hold multiple *trees

**Pull request closed without merging**